src/main/kotlin/nl/info/zac/app/zaak/ZaakKoppelenRestService.kt

```diff
@@ -141,8 +141,8 @@ class ZaakKoppelenRestService @Inject constructor(
                 // As per https://vng-realisatie.github.io/gemma-zaken/standaard/zaken
                 // "deelzaken van deelzaken zijn NIET toegestaan"
                 !this.isDeelzaak && !targetZaak.isIndicatie(ZaakIndicatie.DEELZAAK) &&
-                    // a hoofdzaak cannot become also a deelzaak
-                    !(this.is_Hoofdzaak && targetZaak.isIndicatie(ZaakIndicatie.HOOFDZAAK))
+                    // a hoofdzaak cannot become also a hoofzaak and a deelzaak
```

Comment:
If we are testing for a hoofdzaak why does the line above not start with `this.isHoofdzaak && ..`? I am confused.

Reply:
We removed the `this.isHoofdzaak` check as this breaks when the zaak has no link (is neither a hoofdzaak, nor a deelzaak)

The bad thing here is that we have 3 states and 2 flags to check. It's quite confusing indeed. Perhaps we need a better way to deal with koppelen states.

Reply:
Regarding your remark/comment:

If we add `this.isHoofdzaak` that will break when we want to link two cases that are in both in clean state (neither hoofdzaak, nor deelzaak)

Reply:
Ok! Yes, the business logic is quite complex.. Regarding:

>   // a hoofdzaak cannot become also a hoofzaak and a deelzaak

I don't understand this sentence really. I don't understand the word `also` here and I also don't understand the word `and`. Can we rephrase this a bit maybe to make it clearer? Do you mean here that a zaak which is already a hoofdzaak cannot also become a deelzaak of another zaak?